RecordHub.OrderingService/RecordHub.OrderingService.Infrastructure/Services/OrderingService.cs

```diff
@@ -0,0 +1,62 @@
+﻿using AutoMapper;
+using RecordHub.OrderingService.Application.Services;
+using RecordHub.OrderingService.Domain.Entities;
+using RecordHub.OrderingService.Infrastructure.Data.Repositories;
+using RecordHub.Shared.Enums;
+using RecordHub.Shared.MassTransit.Models.Order;
+using System.Security.Claims;
+
+namespace RecordHub.OrderingService.Infrastructure.Services
+{
+    public class OrderingService : IOrderingService
+    {
+        private readonly IMapper _mapper;
+        private readonly IOrderingRepository _repository;
+
+        public OrderingService(
+            IMapper mapper,
+            IOrderingRepository repository)
+        {
+            _mapper = mapper;
+            _repository = repository;
+        }
+
+        public async Task AddOrderAsync(
+            BasketCheckoutMessage message,
+            CancellationToken cancellationToken = default)
+        {
+            var order = _mapper.Map<Order>(message);
+            order.State = StatesEnum.Submitted;
+
+            await _repository.AddAsync(order, cancellationToken);
+        }
+
+        public async Task ChangeOrderStateAsync(
+            Guid orderId,
+            StatesEnum state,
+            CancellationToken cancellationToken = default)
+        {
+            var order = await _repository.GetAsync(orderId, cancellationToken);
+            if (order == null)
+            {
+                throw new EntityNotFoundException(nameof(orderId));
+            }
+
+            await _repository.UpdateStateAsync(orderId, state, cancellationToken);
+        }
+
+        public Task<IEnumerable<Order>> GetUsersOrdersAsync(
```

Comment:
what about async?)

Reply:
I thought it would be better, when its not creating another state machine when not marked with async

Reply:
Yes, it is really have sense. But why do you choose exactly this method not to be async (f.e. AddOrderAsync)?